### PR TITLE
Add Cloud DMN tab

### LINKED ISSUE #2525: Improve DMN modeling UX for Camunda Cloud
__Is your feature request related to a problem? Please describe.__

In Camunda Cloud ~`1.3.0`~ `1.4.0`, we will support evaluating DMN decisions in Zeebe. Since Zeebe will use a different [DMN engine](https://github.com/camunda-community-hub/dmn-scala) than Camunda Platform and doesn't support all features from the Platform, we should remove some properties to improve the modeling UX.

__Describe the solution you'd like__

In order to improve the DMN UX for Zeebe and the new DMN engine, we should adjust the following parts:
* decisions (in DRG view): 
  * remove the property `Version Tag` - not supported by Zeebe
  * remove the property `History Time to Live` - not supported by Zeebe
  * remove the description of the Id property (similar to the Id property of processes) - in Zeebe the ID will be serialized as `decisionId` (in Zeebe, keys are the identifiers of the record entity) 
  * tracked via https://github.com/bpmn-io/dmn-js-properties-panel/issues/29
* decision table:
  * remove the property to select the `Expression Language` on inputs (i.e. input expressions), input entries, and output entries - Zeebe supports only FEEL
  * remove the property `Input Variable` on inputs - in FEEL, the input value can be accessed by using `?` if needed
  * remove the options `integer` + `long` + `double` in favor of a new option `number` on selecting the `Type` of inputs and outputs - in FEEL, there is only a number type (represented as `BigDecimal`)  
  * add the options `time` + `dateTime` + `dayTimeDuration` + `yearMonthDuration` + `Any` on selecting the `Type` of inputs and outputs - in order to cover the most common types of FEEL
* literal expression:
  * remove the property to select the expression language - Zeebe supports only FEEL 
  * remove the options `integer` + `long` + `double` in favor of a new option `number` on selecting the `Variable Type`  - in FEEL, there is only a number type 
  * add the options `time` + `dateTime` + `dayTimeDuration` + `yearMonthDuration` + `Any` on selecting the `Variable Type` of inputs and outputs - in order to cover the most common types of FEEL

__Describe alternatives you've considered__

No changes. It should be possible to deploy a DMN with the current modeler if no unsupported features are used. Otherwise, the deployment with the DMN is rejected. 

__Additional context__

None.

### PR DESCRIPTION
**Pull request by barmac**
image

Closes #2525 

---

Built on top of https://github.com/camunda/camunda-modeler/pull/2838

The Cloud DMN diagrams can be created via menu; the welcome page button will be implemented via https://github.com/camunda/camunda-modeler/issues/2844

Use [cloud.dmn.txt](https://github.com/camunda/camunda-modeler/files/8351505/cloud.dmn.txt) to open existing file.

Artifacts:

- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/2525-add-cloud-dmn-tab/camunda-modeler-2525-add-cloud-dmn-tab-linux-x64.tar.gz
- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/2525-add-cloud-dmn-tab/camunda-modeler-2525-add-cloud-dmn-tab-mac.dmg
- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/2525-add-cloud-dmn-tab/camunda-modeler-2525-add-cloud-dmn-tab-mac.zip
- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/2525-add-cloud-dmn-tab/camunda-modeler-2525-add-cloud-dmn-tab-win-ia32.zip
- https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/2525-add-cloud-dmn-tab/camunda-modeler-2525-add-cloud-dmn-tab-win-x64.zip

### CODE REVIEW

**Comment by barmac** [comment]
Rebased on `develop` so that you can use the shiny new welcome tab button: https://github.com/camunda/camunda-modeler/issues/2844

File: client/src/app/App.js [hunk 1]
```diff
@@ -1670,6 +1670,10 @@ export class App extends PureComponent {
       return this.createDiagram('cloud-bpmn');
     }
 
+    if (action === 'create-cloud-dmn-diagram') {
```
Comment (pinussilvestrus) [inline client/src/app/App.js:1673]: Should we add [a test here](https://github.com/camunda/camunda-modeler/blob/develop/client/src/app/__tests__/AppSpec.js#L281)?

File: client/src/app/tabs/cloud-dmn/DmnEditor.js [hunk 2]
```diff
@@ -0,0 +1,1023 @@
+/**
```
Comment (pinussilvestrus) [inline client/src/app/tabs/cloud-dmn/DmnEditor.js:1]: The two dmn editors are 99% identical: https://www.diffchecker.com/mzSnaLJT

image

But I guess there is still nothing we can do against (right now)

**Review by pinussilvestrus — COMMENTED** [review]
Currently, there is no real distinction between a C7 and a C8 DMN tab when opening new files. Both show "Camunda Platform" in the status bar. Will this change once we have the engine profile select or is this a bug?

Camunda Platform 7

image


Camunda Platform 8

image

**Comment by barmac** [comment]
> Currently, there is no real distinction between a C7 and a C8 DMN tab when opening new files. Both show "Camunda Platform" in the status bar. Will this change once we have the engine profile select or is this a bug?

This is because under the new name scheme we display "Camunda Platform {version}". Since we don't extract the version yet, it displays only the first part. In the release, we will display both the name and the version.

tldr; not a bug

File: client/src/app/App.js (hunk 1 above)
Comment (barmac) [reply client/src/app/App.js:1673]: 👍

Comment (barmac) [reply client/src/app/App.js:1673]: It's already there: https://github.com/camunda/camunda-modeler/pull/2845/files#diff-b86fdef65a943262634366fd9d7fe4c16c071abbda264ab9aae9f1bc7375c2caR306

😅

File: client/src/app/tabs/cloud-dmn/DmnEditor.js (hunk 2 above)
Comment (barmac) [reply client/src/app/tabs/cloud-dmn/DmnEditor.js:1]: Indeed, it's even worse than with BPMN. Some changes will come when we implement tracking and plugins, but still this is a situation which is far from ideal.

File: client/src/app/App.js (hunk 1 above)
Comment (pinussilvestrus) [reply client/src/app/App.js:1673]: Oh, codecov fooled me then 😄

**Review by pinussilvestrus — APPROVED** [review]
LGTM
